library/src/main/java/org/apache/apex/malhar/lib/window/impl/KeyedWindowedJoinOperatorImpl.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.malhar.lib.window.impl;
+
+import org.apache.apex.malhar.lib.window.ControlTuple;
+import org.apache.apex.malhar.lib.window.JoinWindowedOperator;
+import org.apache.apex.malhar.lib.window.Tuple;
+
+import com.google.common.base.Function;
+
+import com.datatorrent.api.DefaultInputPort;
+import com.datatorrent.api.annotation.InputPortFieldAnnotation;
+import com.datatorrent.lib.util.KeyValPair;
+
+
+/**
+ * Keyed Windowed Join Operator to join two streams of keyed tuple with a key. Please use
+ * {@link WindowedJoinOperatorImpl} for non-keyed joining.
+ *
+ * @param <KeyT> Type of the key used to join two streams.
+ * @param <InputT1> The type of the value of the keyed input tuple from first stream.
+ * @param <InputT2> The type of the value of the keyed input tuple from second stream.
+ * @param <AccumT> The type of the accumulated value in the operator state per key per window.
+ * @param <OutputT> The type of the value of the keyed output tuple.
+ */
+public class KeyedWindowedJoinOperatorImpl<KeyT, InputT1, InputT2, AccumT, OutputT>
```

Comment:
@tweise Sorry I am not quite understand your question. To use `KeyedWindowedJoinOperatorImpl`, user has to specify a key and the joining behavior is defined through the join accumulation the operator is using, it could be more that simply merging two streams together.